numpy/random/tests/test_generator_mt19937.py

```diff
@@ -758,6 +758,12 @@ def test_permutation(self):
         actual = random.permutation(arr_2d)
         assert_array_equal(actual, np.atleast_2d(desired).T)
 
+        bad_x_str = "abcd"
+        assert_raises(np.AxisError, random.permutation, bad_x_str)
+
+        bad_x_float = 1.2
+        assert_raises(np.AxisError, random.permutation, bad_x_float)
+
```

Comment:
This block and the try, except block above should be the only changes in this PR. All the other formatting changes should be reverted.